common/config/persistence.go

```diff
@@ -68,18 +73,25 @@ func (c *Persistence) Validate() error {
 	// Valid dual visibility combinations (order: primary, secondary):
 	// - visibilityStore (advanced sql),  secondaryVisibilityStore (advanced sql)
 	// - visibilityStore (es),            visibilityStore (es) [via elasticsearch.indices config]
+	// - visibilityStore (es),            secondaryVisibilityStore (es)
 	//
 	// Invalid dual visibility combinations:
 	// - visibilityStore (advanced sql),  secondaryVisibilityStore (es)
-	// - visibilityStore (es),            secondaryVisibilityStore (any)
+	// - visibilityStore (es),            secondaryVisibilityStore (advanced sql)
 
 	if c.VisibilityStore == "" {
 		return errors.New("persistence config: visibilityStore must be specified")
 	}
-	if c.DataStores[c.VisibilityStore].Elasticsearch != nil && c.SecondaryVisibilityStore != "" {
+	if c.SecondaryVisibilityStore != "" {
+		if (c.DataStores[c.VisibilityStore].Elasticsearch == nil && c.DataStores[c.SecondaryVisibilityStore].Elasticsearch != nil) ||
+			(c.DataStores[c.VisibilityStore].Elasticsearch != nil && c.DataStores[c.SecondaryVisibilityStore].Elasticsearch == nil) {
+			return ErrInvalidVisibilityStoreCombination
+		}
```

Comment:
Let's make this more readable:
```suggestion
		isPrimaryEs := c.DataStores[c.VisibilityStore].Elasticsearch != nil
		isSecondaryEs := c.DataStores[c.SecondaryVisibilityStore].Elasticsearch != nil
		if (isPrimaryEs != isSecondaryEs) {
			return ErrInvalidVisibilityStoreCombination
		}
```